polytracker/include/taintdag/section.h

```diff
@@ -137,11 +137,13 @@ template <typename T> struct FixedSizeAlloc : SectionBase {
   // N.b. count()/size() competes for the same lock.
   template <typename... Args>
   std::optional<ConstructCtx> construct(Args &&... args) {
-    return map(SectionBase::write(entry_size()), [&](auto &ctx) {
-      return ConstructCtx{.ctx = std::move(ctx),
-                          .t = *new (&*ctx.mem.begin())
+    if (auto write_context = SectionBase::write(entry_size()); write_context) {
```

Comment:
```C
if (auto write_context = SectionBase::write(entry_size())) {...}
```
Should be sufficient, right? Or am I missing something? Same with other changes in this commit.

Reply:
Good catch! Fixed in 58d650f .